Merge Princeton into Cohere

a1icja opened the pull request:
<!--
Thanks for contributing a pull request!
-->

#### Reference Issues/PRs
<!--
Example: "Fixes #1234", "See also #3456"
Please use keywords (e.g., Fixes) to create link to the issues or pull requests
you resolved, so that they will automatically be closed when your pull request
is merged. See https://github.com/blog/1506-closing-issues-via-pull-requests
-->

#### What does this implement/fix? Explain your changes.
<!--
Please include a brief explanation of how your solution
fixes the tagged issue(s), along with what files / entities have
been modified for this fix.
-->

#### Any other comments?

🧡 Thanks for contributing!

klieret committed:
Fix opening PR from fork (#229)

klieret committed:
Fix opening PR from fork

klieret committed:
Add changelog

klieret committed:
Tests to use fast experimental communication strategy (#230)

pre-commit-ci[bot] committed:
chore: update pre-commit hooks (#231)

updates:
- [github.com/astral-sh/ruff-pre-commit: v0.3.5 → v0.3.7](https://github.com/astral-sh/ruff-pre-commit/compare/v0.3.5...v0.3.7)

Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>

john-b-yang committed:
Fix pypi package installation command

john-b-yang committed:
Update to evaluation logic

klieret committed:
Doc: missing 'no' in error message about --open_pr

klieret committed:
Better error handling for --open_pr (#239)

Closes #237

klieret committed:
Speed up testing with persistent containers & remove them end of session (#238)

Closes #228
Closes #201

JGalego committed:
Refactored to AnthropicModel and BedrockModel to avoid code duplication; Added custom error messages

JGalego committed:
Added Claude 3 Opus
https://aws.amazon.com/blogs/aws/anthropics-claude-3-opus-model-on-amazon-bedrock/

JGalego committed:
Fixed model name logic and typing bugs; Added missing return statements

JGalego committed:
Fixed None submission bug

JGalego committed:
Fixed token-counting for older models with Bedrock
https://github.com/anthropics/anthropic-sdk-python/issues/353

DanjieTang committed:
Do not attempt to save patch with empty patch (#242)

* Fixed a potential error

I've ran into this error several times, where it says model_patch can't be None and ending the entire program.

* Do not attempt to save patch with empty patch

---------

Co-authored-by: Kilian Lieret <kilian.lieret@posteo.de>

klieret committed:
Readme: GH token is optional

klieret committed:
Add usage doc to run.py (#243)

klieret committed:
Merge branch 'main' into feat/bedrock-claude-support

klieret committed:
Do not use unix signal calls

klieret committed:
Remove debug print statement with experimental communicate

JGalego committed:
Added max_tokens_to_sample for older models to avoid Bedrock val errors; Changed anthropic_history_to_messages output type

JGalego committed:
Added missing rich_argparse pkg

JGalego committed:
Change from claude 2 to claude 2.0 (see https://github.com/anthropics/anthropic-sdk-python/issues/255)

john-b-yang committed:
Update authors

mspronesti committed:
fix: TARGETARCH not set on some OS/docker setups (#249)

zgrannan committed:
Add GPT4-turbo model (#252)

john-b-yang committed:
Update authors

closji committed:
Add isolated flag to flake8 linting

closji committed:
Add isolated flag to flake8 linting

nfedyashev committed:
Fix typo - "doensn't" in templates (#254)

nfedyashev committed:
Fix typo - "succesfully" in templates (#255)

JGalego committed:
Changed alias name (claude --> claude-2) and target (claude-2.0 --> claude-2.1)

klieret committed:
Catch one more docker error if docker isn't running (#257)

klieret committed:
Refactor run.py main function into class with hook structure (#253)

* WIP

* Refactor run.py into class with hook structure

Closes #170

* Add some more unit tests

* Some more tests

klieret committed:
Merge pull request #207 from JGalego/feat/bedrock-claude-support

feat: Amazon Bedrock support (Claude models)

Borda committed:
pkg: merge all packaging stuff into pyproject.toml (#256)

* pkg: merge all packaging stuff into pyproject.toml

* Add trivial test for packaging

* Add Carlos' email to packaging

---------

Co-authored-by: Kilian Lieret <kilian.lieret@posteo.de>

klieret committed:
Web: Can start runs from flask

klieret committed:
Web: Lay flask scaffolding

klieret committed:
Web: Split feed into two

klieret committed:
Use legacy API for claude-2.1

Thanks to @mikanfactory for spotting this!

klieret committed:
Add hooks to agent (#258)

* Add hooks to agent

* Test hook & fix non-running other tests

ofirpress committed:
Update defaults.sh - scroll_down was misnamed

klieret committed:
Web: Use agent hooks

klieret committed:
Web: Separate messages in feeds; markdown support

klieret committed:
Use a shorter timeout duration for tests (#264)

klieret committed:
WIP

klieret committed:
Adding more hooks to env and agent (#265)

closji committed:
Update defaults and add last_5_history configs

klieret committed:
Web: Add prompts to feed

klieret committed:
Web: Switch to using jquery

klieret committed:
Web: Add step index and scroll to it

pre-commit-ci[bot] committed:
chore: update pre-commit hooks (#268)

updates:
- [github.com/astral-sh/ruff-pre-commit: v0.3.7 → v0.4.1](https://github.com/astral-sh/ruff-pre-commit/compare/v0.3.7...v0.4.1)

Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>

klieret committed:
Web: Moved most of the interface to react

waterson committed:
Pass Python version to get_environment_yml

This ensures that the `environment.yml` is correctly constructed with the
specific Python version required for the instance.

klieret committed:
Web: Bring back highlighting

tamdogood committed:
minor changes for server and client endpoints to better handling cors

closji committed:
Update swe_env.py

Replicates installation behavior from SWE-bench at https://github.com/princeton-nlp/SWE-bench/blob/cfb20092bbbee9683176177b2f59b85f522e7f27/swebench/harness/context_manager.py#L354-L376

klieret committed:
Web Fix: Every message to appear only once

john-b-yang committed:
Merge pull request #271 from waterson/waterson/python-version-environment-yml

Pass Python version to get_environment_yml

john-b-yang committed:
Minor condition changes

klieret committed:
Web feat: Restore scrolling behavior

ofirpress committed:
Update edit_linting.sh - fix grammar issue

ofirpress committed:
Update cursors_edit_linting.sh - fix grammar issue

klieret committed:
Web feat: Kill running computation

klieret committed:
Web: Rename folder web -> api

klieret committed:
Web: Remove files from flask prototype

klieret committed:
Web refactor: Split up server.py

klieret committed:
Web feat: Display log messages (partially broken)

Unfortunately all threads share the same stdout, so it's not trivial at
all to redirect different threads to different stdouts

mikanfactory committed:
Fix Together model validation error (#236)

* test: add unit test for Together model

* fix: deal with the new Together API

* chore: specify together version

* refactor: clean code

* change together model versioning from ">=~" to ">=" and write comment

* raise exception when together SDK version is below 1.1.0

* refactor: update unit test format

* speficy max_tokens

klieret committed:
Web enh: Control button activity depending on run state

klieret committed:
Web enh: Auto-scroll log messages

klieret committed:
Web enh: Only scroll and highlight after computation is finished

klieret committed:
Web enh: Make sure that killing thread succeeds

klieret committed:
Web: Factor out Feed.js; fix highlighting of step == null

klieret committed:
Web WIP: Started to integrate swe-agent/demo parts

klieret committed:
Web WIP: Styling and refactoring

klieret committed:
Web WIP: Split up message types

klieret committed:
Web enh: Bring in some highlighting

klieret committed:
Web feat: Include the rest of the demo code

tamdogood committed:
minor refactor of the server to fix 403 code and also missing secret_key

tamdogood committed:
adding requirements.txt there are many version conflicts in the codebase, it's hard to run the server without having the correct version. Adding the requirements to standardize the future setup

pre-commit-ci[bot] committed:
chore: update pre-commit hooks (#282)

updates:
- [github.com/astral-sh/ruff-pre-commit: v0.4.1 → v0.4.2](https://github.com/astral-sh/ruff-pre-commit/compare/v0.4.1...v0.4.2)

Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>

klieret committed:
WIP: Create GH codespaces

klieret committed:
Codespaces: Fix permissions for talking to docker daemon

klieret committed:
Codespaces: Pull swe-agent image; conda init

klieret committed:
Codespace: Automatically activate swe-agent env

klieret committed:
Codespaces: Fix: don't overwrite bashrc (#288)

[Skip-ci]

ofirpress committed:
Update README.md

klieret committed:
Codespaces: Run additional setup as onCreateCommand

ofirpress committed:
Update devcontainer.json

klieret committed:
Revert "Update devcontainer.json"

This reverts commit c8542e7e3c2c46d84caaa1890943d7181cd40dc6.

klieret committed:
Add helpful message about conda env activation (#289)

klieret committed:
Codespaces: Use pip install instead of creating new conda env (#291)

klieret committed:
Doc: Avoid invalid github token (#292)

[skip-ci]

klieret committed:
Improve codespace setup & documentation (#293)

[skip-CI]

* Codespaces: Remove shell setting; fix extensions setting

[skip-ci]

* Codespaces: Copy sample keys.cfg

[skip-ci]

* Codespaces: Add codespace badge

[skip-CI]

klieret committed:
Doc: Add codespace video

klieret committed:
Codespace: Add startup message to terminal (#294)

[skip-ci]

klieret committed:
Web: Fix port of server for websocket

klieret committed:
Web: Redirect all relevant stderr & handle errors in thread

klieret committed:
Web: Rename feeds

klieret committed:
Web: Add warning message if server is not connected

klieret committed:
Merge branch 'main' into web-ui-dev

klieret committed:
Web: Simple script to start web server

klieret committed:
Codespace: Install npm

klieret committed:
Web: Make sure that pm2 is found in cleanup method

klieret committed:
CI: Use pip for installation instead conda (#299)

* CI: Use pip for installation instead conda

* Make sure that python is set up

klieret committed:
docker ignore everything from gitignore

[skip-ci]

klieret committed:
Setup: do not duplicate requirements (#300)

* WIP

* Fix: Need to copy app first before pip install .

klieret committed:
CI: Add GHA to test running setup.sh (#302)

klieret committed:
Fix readme badge links (#303)

klieret committed:
Web: Factor out run control

klieret committed:
Enh: Allow to directly specify problem statement (#308)

klieret committed:
Merge branch 'main' into web-ui-dev

klieret committed:
Web: Allow different ways to specify PS; repo path; bootstrap

klieret committed:
Web: Place controls in accordion

klieret committed:
Web: Format test run checkbox as switch

klieret committed:
Web fix: Reset highlighted step after running

klieret committed:
Web: Add flask dependencies

vilkinsons committed:
fix:typo

ofirpress committed:
Merge pull request #310 from hashdeps/main

fix: typo in root readme

klieret committed:
Fix: Include demonstrations in dockerignore (#311)

[skip-ci]

tamdogood committed:
disabled bubbles' scrolling and text color

tamdogood committed:
Rearranged input elements

tamdogood committed:
removed unnecessary elements

tamdogood committed:
create copy function for log panel

tamdogood committed:
change color for highlighted messages

klieret committed:
Web: Replace accordion with tabs

klieret committed:
Web: Various Styling improvements

klieret committed:
Web fix: Checkbox default state not reflected

klieret committed:
Web fix: Highlighting in terminal (restore linebreaks)

klieret committed:
Web enh: Remove highlight if mouse leaves message

klieret committed:
Web enh: Add timeout to highlight/scroll

ofirpress committed:
Update README.md

klieret committed:
Web enh: Run button layout; logo; remove header

klieret committed:
Web: Add link to github readme

klieret committed:
Web feat: Model selection

klieret committed:
Web enh: Fix spacing of code blocks

klieret committed:
Better messages for InstantEmptySubmitTestModel

klieret committed:
Web: Remove "Thought" and fix info msg styling

klieret committed:
Web enh: Add start message; style no connection error msg

klieret committed:
Web style: Remove three dots; move logos into window bars

klieret committed:
Web style: Descriptions for other text fields

klieret committed:
Web ref: Move CSS to appropriate files

klieret committed:
Web: Move swe-agent logo to top bar

klieret committed:
Web: Font-size adjustments

klieret committed:
Web: Minimize menu when run started

klieret committed:
Web: Only show "Copy to clipboard" after run

klieret committed:
Web: Show critical errors in top banner

klieret committed:
Web: Show explicit support for local PS or repos

klieret committed:
Web: Improve handling of container closing

klieret committed:
Web: Assume compute has finished when 20s no update

klieret committed:
Web: Always use experimental speedups

pre-commit-ci[bot] committed:
chore: update pre-commit hooks (#318)

updates:
- [github.com/astral-sh/ruff-pre-commit: v0.4.2 → v0.4.3](https://github.com/astral-sh/ruff-pre-commit/compare/v0.4.2...v0.4.3)

Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>

klieret committed:
Web: Add note about successful pitch; real example by default

klieret committed:
Web: Catch bug with empty observation

Borda committed:
lint: use `typos` as precommit's hook (#259)

* lint: use typos as precommit hook

* fixing typos

klieret committed:
Doc: Recommend pip install instead of conda (#304)

* Doc: Recommend pip install instead of conda

* Fix numbering

[skip-ci]

* Doc: Make installation with pip the default

klieret committed:
Web: Reformat code with prettier

klieret committed:
Doc fix: Misleading comment about env vars with docker

RainRat committed:
Update swe_env.py

fix typo

RainRat committed:
Merge branch 'main' of https://github.com/RainRat/SWE-agent

klieret committed:
Print helpful error message when flask isn't available

klieret committed:
Comment out all keys in sample keys.cfg by default

john-b-yang committed:
Merge pull request #320 from RainRat/main

fix typo in swe_env.py

klieret committed:
Close environment when raising exception

klieret committed:
Web: Always raise exceptions

klieret committed:
Web: Switch to silver logos

klieret committed:
Web: Change title of agent feed

klieret committed:
Web feat: Allow to specify python version & req pkgs

klieret committed:
Web feat: Allow to specify path to shell script

klieret committed:
Web: Temporarily disable timeout-based setIsComputing

klieret committed:
Web feat: Set custom install command

klieret committed:
Doc: Fix links to installation issues section

[skip-ci]

klieret committed:
Doc: Fix link to installation issues section

klieret committed:
Web style fix: Position of logo for narrow screens

klieret committed:
Fix: Handling of long problem statements

klieret committed:
Style: Black format api code

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

klieret committed:
Remove typo/comment

klieret committed:
Fix: Handling gh issue URLs as problem statements

klieret committed:
Web interface 🎉 (#321)

Web interface 🎉

klieret committed:
Doc: Add gif of web interface

[skip ci]

klieret committed:
Fix typo

[skip ci]

klieret committed:
Fix: Catch container not found and retry after wait

Fixes #322

klieret committed:
Doc: Add web UI instructions

[skip ci]

klieret committed:
Doc: Add information of how to open correct browser window (#324)

[skip ci]

klieret committed:
Doc: Suggest starting web UI in GH codespaces

ofirpress committed:
Update README.md - slight rewording of a header

klieret committed:
Web: Fix script_path input (#334)

Closes #333

[skip ci]

ofirpress committed:
Update README.md - updating bibtex

ofirpress committed:
Update README.md

ofirpress committed:
Update README.md

klieret committed:
Readme: Fix links

[skip ci]

klieret committed:
Improve handling of incorrect repo_path configs (#340)

klieret committed:
Always get base_commit hash (can be specified as tag/branch) (#341)

klieret committed:
Fix: Don't print patch msg for exit_cost patch (#343)

Closes #342

raymyers committed:
Add gpt-4o model (#344)

Co-authored-by: Ray Myers <rmyers@indeed.com>

klieret committed:
Fix: Do not request job control in bash (#345)

Closes #331

It's unlikely that job control was ever granted. Currently we're getting

ERROR    Unexpected container setup output: /bin/bash: cannot set terminal process group (-1): Inappropriate ioctl for device
         /bin/bash: no job control in this shell

Because of this.

klieret committed:
Fix: --base_commit not used for gh urls (#346)

pre-commit-ci[bot] committed:
chore: update pre-commit hooks (#347)

updates:
- [github.com/crate-ci/typos: v1.20.7 → v1.21.0](https://github.com/crate-ci/typos/compare/v1.20.7...v1.21.0)
- [github.com/astral-sh/ruff-pre-commit: v0.4.3 → v0.4.4](https://github.com/astral-sh/ruff-pre-commit/compare/v0.4.3...v0.4.4)
- [github.com/pre-commit/mirrors-prettier:  → v4.0.0-alpha.8](https://github.com/pre-commit/mirrors-prettier/compare/...v4.0.0-alpha.8)

Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>

klieret committed:
Fix: Separate data path/traj dir cause exception (#348)

john-b-yang committed:
Readme: Shorten ACI text

[skip ci]

ofirpress committed:
Update README.md

ofirpress committed:
Update README.md

klieret committed:
Remove duplicated abstract method (#355)

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/cohere-ai/SWE-agent?pullRequest=1) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/cohere-ai/SWE-agent?pullRequest=1) before we can accept your contribution.<br/>**0** out of **4** committers have signed the CLA.<br/><br/>:x: john-b-yang<br/>:x: klieret<br/>:x: nims11<br/>:x: ofirpress<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/cohere-ai/SWE-agent?pullRequest=1) it.</sub>

klieret committed:
Web: Refactor state into one runConfig with use-immer (#350)

klieret committed:
Web: Allow to specify commit hash (#358)

Closes #336

klieret committed:
CI: Use uv pip install (#360)

* CI: Use uv pip install

* CI: Try with explicit virtuale_env

klieret committed:
Web: Shorten long error messages in banner (#361)

Closes #330

klieret committed:
Wait longer if processes still running (#364)

Closes #363

ofirpress committed:
Update default_sys-env_cursors_window100-detailed_cmd_format-full_history-1_demos.yaml - adding warning to experimental config

ofirpress committed:
Update default_sys-env_cursors_window100-detailed_cmd_format-last_5_history-1_demos.yaml - adding warning to experimental config

ofirpress committed:
Update xml_sys-env_cursors_window100-detailed_cmd_format-full_history-1_demos.yaml - adding warning to experimental config

ofirpress committed:
Update xml_sys-env_cursors_window100-detailed_cmd_format-last_5_history-1_demos.yaml - adding warning to experimental config

ofirpress committed:
Update README.md - clarify that traj arg has to be absolute path

nims11 committed:
Fix handling of not_generated/no_generation in inspector (#332)

* Fix typo in inspector server.py

This leads to "Results format not recognized" error whenever viewing the eval report for a trajectory.

* Fix: Consistently handle no_generation vs not_generated

---------

Co-authored-by: Kilian Lieret <kilian.lieret@posteo.de>

klieret committed:
Inspector: Better labels for roles (#368)

Closes #365

klieret committed:
Change icons for trajectory viewer (#370)

Closes #365

klieret committed:
Move documentation to mkdocs (#371)

klieret committed:
Docs: Add installation overview page (#377)

klieret committed:
Docs: Add github button; edit feature

klieret committed:
Docs: Change color preferences

klieret committed:
Docs: Add next prev/buttons

klieret committed:
CI: Skip CI for PRs that only touch docs

klieret committed:
Docs: Switch to documentation

klieret committed:
Add default environment_setup config (#351)

[skip ci]

klieret committed:
Docs: Fix max-width tag of doc link

[skip ci]

klieret committed:
Doc: Significantly expand CL tutorial

klieret committed:
Doc: Restore docs on starting web UI on GH codespaces

klieret committed:
Doc: Add copy button; highlight specific lines

klieret committed:
Doc/CI: Speed up documentation build

klieret committed:
Doc: Move config docs to mkdocs

klieret committed:
CI: Set VIRTUAL_ENV for uv

klieret committed:
Doc: Fix inclusion of image in config.md

klieret committed:
Doc: Attempt to use relative image path

klieret committed:
Doc: Add changelog

Closes #335

klieret committed:
Docs: Add more READMEs to mkdocs

klieret committed:
Remind people not to use screenshots when reporting bugs

klieret committed:
Remind people not to use screenshots for error messages

klieret committed:
Upper bound request version to avoid docker-py bug (#390)

Closes #379

klieret committed:
Doc: Replace symlinks with markdown files with links (#392)

Closes #388

klieret committed:
Docs: Add search (#393)

Closes #387

Search is added by default but must be manually added if any other plugins are
configured

See https://github.com/squidfunk/mkdocs-material/blob/master/docs/setup/setting-up-site-search.md

klieret committed:
Docs: Add code of conduct (#394)

[skip ci]

klieret committed:
Add nodejs to swe-agent-run container (#396)

klieret committed:
Docs: Note about old images from the hub (#395)

klieret committed:
Docs: Advice to update pip if unsuccessful (#399)

klieret committed:
Show error log if web server fails (#400)

[skip ci]

klieret committed:
CI: Fix passing python path to uv (#401)

klieret committed:
Docs: Detailed way to start the web server (#402)
